### PR TITLE
use change address for paypro refund

### PR DESCRIPTION
**Pull request by matiu**
fixes: https://github.com/bitpay/copay/issues/3582

### CODE REVIEW

File: test/client.js [hunk 1]
```diff
@@ -1924,6 +1924,8 @@ describe('client API', function() {
     it('Should send correct refund address', function(done) {
       clients[0].getTxProposals({}, function(err, txps) {
         should.not.exist(err);
+        var changeAddress = txps[0].changeAddress.address;
+        console.log('[client.js.1927:changeAddress:]', changeAddress); //TODO
```
Comment (isocolsky) [inline test/client.js]: Please remove :)

**Comment by isocolsky** [comment]
LGTM

**Comment by matiu** [comment]
updated, thanks!

**Comment by dabura667** [comment]
This will break portability to other apps, as internal chains are expected to have a gap of 0 (they are only used for change, which is only used when an actual output will be assigned to it.

**Comment by matiu** [comment]
In case of a refund, the changeAddress will be actually be reused.  Unluckily, unless we extended BIP44 with a new branch for refund addresses that allow huge gaps, I dont see an other solution.

**Pull request merged**